File: week01/Theme02/Theme02_Code06.cpp
// Theme02_Code06.cpp

#include <iostream>
#include <queue>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	stack<int> s;

	s.push(0);
	s.push(1);
	s.push(2);

	cout << s.size() << endl; // 3
	cout << s.empty() << endl; // 0
	
	s.push(3);
	s.push(4);
	s.pop();
	s.pop();

	cout << s.size() << endl; // 3
	cout << s.empty() << endl; // 0

	s.pop();
	s.pop();
	s.pop();
	
	cout << s.size() << endl; // 0
	cout << s.empty() << endl; // 1
}

File: week01/Theme02/Theme02_Code06.py
# Theme02_Code06.py

from collections import deque

d = deque()

d.appendleft(0)
d.appendleft(1)
d.append(2)
d.append(3)
d.appendleft(4)

print(d[0], d[-1]) # 4 3

d.popleft()
d.pop()
d.appendleft(5)
d.append(6)

print(d[0], d[-1]) # 5 6